File: KH.Dto/Lookups/PermissionsDto/Response/PermissionResponse.cs
using KH.Domain.Entities;

namespace KH.Dto.Lookups.PermissionsDto.Response;

public class PermissionResponse
{
  public long Id { get; set; }
  public string NameEn { get; set; }
  public string NameAr { get; set; }
  public bool Disabled { get; set; }
  public bool Checked { get; set; }
  public List<PermissionResponse> Childrens { get; set; } = new List<PermissionResponse>();
  public int SortKey { get; set; }
  public long? ParentId { get; set; }

  public PermissionResponse()
  {


  }

  public PermissionResponse(Permission e)
  {
    Id = e.Id;
    NameAr = e.NameAr;
    NameEn = e.NameEn;
    SortKey = e.SortKey;
    ParentId = e.ParentId;
    Childrens = e.Children.Select(x => new PermissionResponse(x)).ToList();

  }
}

File: KH.WebDashboard/src/app/@core/models/responses/permission-response.ts



export class PermissionResponse {
  id: number;
  key: string;
  nameEn: string;
  nameAr: string;
  disabled: boolean;
  checked: boolean;
  childrens: PermissionResponse[];
  sortKey: number;
  parentId: number | null;
  roleId!: number | null; // Added property to hold roleId later on


  constructor(data: Partial<PermissionResponse>) {
    this.id = data.id || 0;
    this.key = data.key || '';
    this.nameEn = data.nameEn || '';
    this.nameAr = data.nameAr || '';
    this.disabled = data.disabled || false;
    this.checked = data.checked || false;
    this.childrens = (data.childrens || []).map(child => new PermissionResponse(child));
    this.sortKey = data.sortKey || 0;
    this.parentId = data.parentId || null;
  }
}
